src/Compiler/TypedTree/TypedTreeOps.fs

```diff
@@ -10407,3 +10407,17 @@ let (|EmptyModuleOrNamespaces|_|) (moduleOrNamespaceContents: ModuleOrNamespaceC
         else
             None
     | _ -> None
+
+let (|TTypeMultiDimensionalArrayAsGeneric|_|) (t: TType) =
+    let rec (|Impl|_|) t =
+        match t with
```

Comment:
This code is wrong 

1. once again we're matching directly on TType, we should **almost never** be doing this
2. it looks like it is detecting jagged arrays, which is wrong

Reply:
Will have a look next week